constraint-solver/src/solver.rs

```diff
@@ -155,7 +156,19 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq + Display + Debug> Solver<T, V>
             }
             Effect::BitDecomposition(..) => unreachable!(),
             Effect::Assertion(..) => unreachable!(),
-            Effect::ConditionalAssignment { .. } => todo!(),
+            Effect::ConditionalAssignment {
+                variable,
+                condition: Condition { value, condition },
+                in_range_value,
+                out_of_range_value,
+            } => {
+                let value = value.try_to_number().unwrap();
```

Comment:
But if `value` is a concrete number, why don't we return an unconditional assignment right away in `combine_to_conditional_assignment`?

Reply:
Ah ok, in our case, `first_assignment` and `second_assignment` will always be numbers.

I would still prefer to pull that into `combine_to_conditional_assignment`